### PR TITLE
TF-2461 Impossible to refresh an expired JWT claim

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2461

## Root cause

- Retry on `onError` is blocked by queue interceptor, the current one was not resolved but still try to get the new one and the new one also have the error too.

### CODE REVIEW

File: lib/features/login/data/network/config/time_out_interceptors.dart [hunk 1]
```diff
@@ -1,22 +0,0 @@
-import 'package:core/utils/app_logger.dart';
-import 'package:core/utils/platform_info.dart';
-import 'package:dio/dio.dart';
-import 'package:tmail_ui_user/features/login/data/network/endpoint.dart';
-
-class TimeOutInterceptors extends InterceptorsWrapper {
```
Comment (hoangdat) [inline lib/features/login/data/network/config/time_out_interceptors.dart:6]: why we need to remove it?

Comment (dab246) [reply lib/features/login/data/network/config/time_out_interceptors.dart:6]: IMO, it's not necessary

File: test/features/interceptor/authorization_interceptor_test.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            requestOptions: RequestOptions(path: '/jmap')));
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        expect(authorizationInterceptors.validateToRefreshToken(dioError401), true);
+      });
+
+      test('validateToRefreshToken should return false when conditions are not met', () async {
```
Comment (hoangdat) [inline test/features/interceptor/authorization_interceptor_test.dart:72]: we found 4 conditions in checking refresh token, each condition should have a testcase

File: test/features/interceptor/authorization_interceptor_test.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      tokenOidcCacheManager,
+      accountCacheManager,
+      iosSharingManager);
+
+    dio.interceptors.add(authorizationInterceptors);
+
+    dioAdapter = DioAdapter(dio: dio);
+  });
+
+  group('AuthorizationInterceptor test', () {
```
Comment (hoangdat) [inline test/features/interceptor/authorization_interceptor_test.dart:56]: How about to add more test-case:
- related to `FileUploader.uploadAttachmentExtraKey`
- make sure we should not be blocked anymore

Comment (dab246) [reply test/features/interceptor/authorization_interceptor_test.dart:56]: - In the case of uploading it depends too much on external objects, it is difficult to write test logic here. We have to separate it into separate functions to make it easier to write unit tests.
- We've got a `refreshingTokensOIDC` test case that only calls once when the token expires.

Comment (hoangdat) [reply test/features/interceptor/authorization_interceptor_test.dart:56]: - ok for the case of `FileUploader.uploadAttachmentExtraKey`
- but, it is `QueueInterceptor`, so I think we must find the way to test the integration of interceptors, make sure it not be blocked in onError anymore

**Comment by dab246** [comment]
We close this PR to reopen new PR (#2508)  to maintain for `v0.11.3`.

**Pull request closed without merging**